numpy/core/tests/test_arrayprint.py

```diff
@@ -395,6 +395,37 @@ def test_wide_element(self):
             "[ 'xxxxx']"
         )
 
+    def test_multiline_repr(self):
+        class MultiLine:
+            def __repr__(self):
+                return "Line 1\nLine 2"
+
+        a = np.array([[None, MultiLine()], [MultiLine(), None]])
+
+        assert_equal(
+            np.array2string(a),
+            '[[None Line 1\n'
+            '       Line 2]\n'
+            ' [Line 1\n'
+            '  Line 2 None]]'
+        )
+        assert_equal(
+            np.array2string(a, max_line_width=5),
+            '[[None\n'
+            '  Line 1\n'
+            '  Line 2]\n'
+            ' [Line 1\n'
+            '  Line 2\n'
+            '  None]]'
+        )
+        assert_equal(
+            np.array_repr(a),
+            'array([[None, Line 1\n'
+            '              Line 2],\n'
+            '       [Line 1\n'
+            '        Line 2, None]], dtype=object)'
+        )
+
```

Comment:
Since it's the motivating use case, would also be nice to have:
```
    def test_nested_array_repr(self):
        arr = np.empty((2, 2), dtype=object)
        arr[0, 0] = np.eye(2)
        arr[0, 1] = np.eye(3)
        arr[1, 0] = None
        arr[1, 1] = np.ones((3, 1)
        assert_equal(
            repr(arr),
            ...   # put whatever you actually get here.
        )
```

Reply:
Done